definitions/declarations/httparchive.js

```diff
@@ -1,17 +1,41 @@
-const stagingTables = ['pages', 'requests', 'parsed_css']
-for (const table of stagingTables) {
+// Staging tables source: https://github.com/HTTPArchive/crawl/blob/main/crawl.py
+['pages', 'requests', 'parsed_css'].forEach(table =>
   declare({
     schema: 'crawl_staging',
     name: table
   })
-}
+)
 
-declare({
-  schema: 'wappalyzer',
-  name: 'technologies'
-})
+assert('corrupted_technology_values')
+  .tags(['crawl_complete'])
+  .query(ctx => `
+SELECT
+  date,
+  client,
+  tech,
+  COUNT(DISTINCT page) AS cnt_pages,
+  ARRAY_AGG(DISTINCT page LIMIT 3) AS sample_pages
+FROM ${ctx.ref('crawl_staging', 'pages')} AS pages
+LEFT JOIN pages.technologies AS tech
+LEFT JOIN tech.categories AS category
+WHERE
+  date = '${constants.currentMonth}' AND
+  (
+    tech.technology NOT IN (SELECT DISTINCT name FROM wappalyzer.technologies)
+    OR category NOT IN (SELECT DISTINCT name FROM wappalyzer.categories)
```

Comment:
Are the `wappalyzer` tables updated with every crawl? Is that planned to be continued after we remove the legacy tables? (I thought this WAS a legacy table!)

Reply:
Those are updated with every repo update, see https://github.com/HTTPArchive/wappalyzer/blob/main/.github/workflows/upload.yml.

These are not dependent on crawl detections.
But they should represent the crawl configuration considering we don't merge changes during crawl.

Reply:
Any risk that a technology that is removed (or renamed) mid-crawl causes it to be deleted? I try not to merge things mid-crawl, but stranger things have happened. But not sure how to deal with that to be honest...

Reply:
Sure. But I think monthly manual changes pose a bigger risk.

- I hope 🤞🏽 we'll find a solution to avoid corrupted values on wptagent during the next attempt.
- To avoid retrospective issues I expect we'll be able to extend `reports.cwv_tect_technologies` with clean historical data.
- We could also backup the original data for N months just in case.